File: backoffice/src/diningTable/domain/exceptions/InvalidDiningTableNumberException.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import DomainException

"""
 *
 * Class 
 *
"""

class InvalidDiningTableNumberException( DomainException ):

    """
     *
     * Constants 
     *
    """

    __INVALID_DINING_TABLE_NUMBER = 146

    """
     *
     * Methods 
     *
    """

    def __init__( self, number : int ) -> None:
        super().__init__(
            self.__INVALID_DINING_TABLE_NUMBER,
            'The dining table number {} is invalid'.format( number )
        )

File: preparation/src/diningTable/domain/exceptions/InvalidDiningTableNumberException.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class InvalidDiningTableNumberException : DomainException {

        /* 
         * 
         * Constants
         *
        */

        private const int INVALID_DINING_TABLE_NUMBER = 146;

        /* 
         * 
         * Methods
         *
        */

        public InvalidDiningTableNumberException( int number ) : base( 
            INVALID_DINING_TABLE_NUMBER,
            string.Format( "The dining table number {0} is invalid", number )
        ) {
        }

    }

}